providers/universal-provider/src/UniversalProvider.ts

```diff
@@ -136,19 +139,40 @@ export class UniversalProvider implements IUniversalProvider {
   }
 
   public async pair(pairingTopic: string | undefined): Promise<SessionTypes.Struct> {
-    const { uri, approval } = await this.client.connect({
-      pairingTopic,
-      requiredNamespaces: this.namespaces,
-      optionalNamespaces: this.optionalNamespaces,
-      sessionProperties: this.sessionProperties,
-    });
+    this.shouldAbortPairingAttempt = false;
+    let pairingAttempts = 0;
+    do {
+      if (this.shouldAbortPairingAttempt) {
+        throw new Error("Pairing aborted");
+      }
 
-    if (uri) {
-      this.uri = uri;
-      this.events.emit("display_uri", uri);
-    }
-    this.session = await approval();
-    this.createProviders();
+      if (pairingAttempts >= this.maxPairingAttempts) {
+        throw new Error("Max auto pairing attempts reached");
+      }
+
+      const { uri, approval } = await this.client.connect({
+        pairingTopic,
+        requiredNamespaces: this.namespaces,
+        optionalNamespaces: this.optionalNamespaces,
+        sessionProperties: this.sessionProperties,
+      });
+
+      if (uri) {
+        this.uri = uri;
+        this.events.emit("display_uri", uri);
+      }
+
+      await approval()
+        .then((session) => {
+          this.session = session;
+        })
+        .catch((error) => {
+          if (error.message !== "Proposal expired") {
```

Comment:
Wish we could use the constant you defined in the other pkg here so it's not a magic string but just not worth the extra PITA to have it in a shared pkg.

Reply:
I tried importing `PROPOSAL_EXPIRY_MESSAGE` earlier and it wasn't working for some reason but now rebuilt everything and it does :100: